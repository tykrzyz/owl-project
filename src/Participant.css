.participant{
    display: flex;
    margin-bottom: 10px;
}

.participant h2{
    font-size: 20px;
    margin: 0px;
}

.participant p{
    font-size: 18px;
    margin: 0;
    font-weight: 400;
}

.info{
    flex: 1;
}

.avatar img{
    width: 40px;
    height: 40px;
    border-radius: 20px;
    flex: 1;
}