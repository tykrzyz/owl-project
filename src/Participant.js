import './Participant.css'
export default function Participant(props){
    let stageStatus = props.onStage ? 'on stage' : 'in session'
    return(
        <div className='participant'>
            <div className='avatar'>
                <img src={props.avatar} alt={props.name + `'s avatar`} />
            </div>
            <div className='info'>
                <h2>{props.name}</h2>
                <p>{stageStatus}</p>
            </div>
        </div>
    )
}