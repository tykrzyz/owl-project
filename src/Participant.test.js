import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';

const participant = {
    id: 1,
    name: 'Koren Templeton',
    avatar:
        'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true
}

describe('<Participant />', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participant {...participant}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<Participant {...participant}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
        });
})