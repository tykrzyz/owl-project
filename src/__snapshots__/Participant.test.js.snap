// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> renders the UI as expected 1`] = `
<div
  className="participant"
>
  <div
    className="avatar"
  >
    <img
      alt="Koren Templeton's avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="info"
  >
    <h2>
      Koren Templeton
    </h2>
    <p>
      on stage
    </p>
  </div>
</div>
`;
